cpp/src/gandiva/interval_holder_test.cc

```diff
@@ -0,0 +1,423 @@
+  EXPECT_EQ(response, 7);
+
+  data = "P10W";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, 70);
+
+  execution_context_.Reset();
+}
+
+TEST_F(TestIntervalHolder, TestUsingCompleteFormatterForCastIntervalDay) {
+  std::shared_ptr<IntervalDaysHolder> interval_holder;
+
+  auto status = IntervalDaysHolder::Make(0, &interval_holder);
+  EXPECT_EQ(status.ok(), true) << status.message();
+
+  auto& cast_interval_day = *interval_holder;
+
+  bool out_valid;
+  std::string data("1742461111");
+  int64_t response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  int64_t qty_days_in_response = 20;
+  int64_t qty_millis_in_response = 14461111;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  data = "P1Y1M1DT1H1M1S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  qty_days_in_response = 1;
+  qty_millis_in_response = 3661000;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  data = "PT48H1M1S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  qty_days_in_response = 2;
+  qty_millis_in_response = 61000;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  data = "PT1S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  qty_days_in_response = 0;
+  qty_millis_in_response = 1000;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  data = "P10DT1S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  qty_days_in_response = 10;
+  qty_millis_in_response = 1000;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  execution_context_.Reset();
+
+  data = "P0DT0S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  qty_days_in_response = 0;
+  qty_millis_in_response = 0;
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+  EXPECT_EQ(response, (qty_millis_in_response << 32) | qty_days_in_response);
+
+  execution_context_.Reset();
+}
+
+TEST_F(TestIntervalHolder, TestUsingCompleteFormatterForCastIntervalYear) {
+  std::shared_ptr<IntervalYearsHolder> interval_years_holder;
+
+  auto status = IntervalYearsHolder::Make(0, &interval_years_holder);
+  EXPECT_EQ(status.ok(), true) << status.message();
+
+  auto& cast_interval_years = *interval_years_holder;
+
+  bool out_valid;
+  std::string data("65851111");
```

Comment:
this looks a bit strange and confusing. If we are expecting input in months (and not millis), the comments in the operator above should be corrected. Also, if we expect months, something smaller like 623 months etc might be better as this looks like a large value for months.

Reply:
The comments were wrong, the input is the number of months. I fixed the comment in the function.

Reply:
ok